gomock/argument_captor.go

```diff
@@ -0,0 +1,40 @@
+package gomock
+
+// ArgumentCaptor is a struct that composes a Matcher, and extends it by storing given arguments in the values slice.
+type ArgumentCaptor struct {
+	Matcher
+	values []interface{}
+}
+
+// Matches method overrides the Matcher.Matches method
+// First it appends any argument(s) used to the values slice.
+// Then the parent Matches method is called.
+func (ac *ArgumentCaptor) Matches(x interface{}) bool {
+	ac.values = append(ac.values, x)
+	return ac.Matcher.Matches(x)
+}
+
+// LastValue returns the last argument the matcher was called with as an interface{}.
+// If the matcher was never called, nil is returned.
+func (ac *ArgumentCaptor) LastValue() interface{} {
+	if len(ac.values) < 1 {
+		return nil
+	}
+	return ac.values[len(ac.values)-1]
+}
+
+// Values returns the all arguments the matcher was called with as a []interface{}.
+// The values are ordered from first called to last called.
+func (ac *ArgumentCaptor) Values() []interface{} {
+	return ac.values
+}
+
+// Captor is a helper method that returns a new *ArgumentCaptor struct with Matcher set to the given matcher m
+func Captor(m Matcher) *ArgumentCaptor {
+	return &ArgumentCaptor{Matcher: m}
+}
+
+// AnyCaptor is a helper method that returns a new *ArgumentCaptor struct with the matcher set to an anyMatcher
+func AnyCaptor() *ArgumentCaptor {
```

Comment:
Thoughts on `CaptureAny()`?

Reply:
This would be ok with me if the call usually happened in the `EXPECT` line, but because captors aren't that useful unless they're declared in advance, I think it makes more sense to keep it as is and view it like a constructor. 

Here's an example:
```go
// here we don't have a way to grab the values
mockDao.EXPECT().InsertIDs(gomock.CaptureAny()).Times(1)

// we need to store the captor first 
idCaptor := gomock.AnyCaptor()
mockDao.EXPECT().InsertIDs(idCaptor).Times(1)
actualIDs := idCaptor.LastValue().([]int)
```